src/main/java/net/minecraftforge/client/event/EntityRenderersEvent.java

```diff
@@ -144,4 +150,36 @@ public EntityModelSet getEntityModels()
             return entityModels;
         }
     }
+
+    /**
+     * An event to provide a safe place to register entity animations
+     */
+    public static class AddAnimations extends EntityRenderersEvent
+    {
+        private final Map<EntityType<?>, List<Consumer<LivingEntityRenderer<?, ?>>>> animations = new HashMap<>();
+
+        /**
+         * Prepares an animation to be added to the given entity type. Animations are registered when
+         * the resource manager is reloading.
+         *
+         * @param type the entity type
+         * @param animation an animation implementation that matches the entity type
+         * @param <T> an entity that extends living entity
+         */
+        @SuppressWarnings("unchecked")
+        public <T extends LivingEntity> void addAnimation(EntityType<T> type, EntityAnimation<T> animation)
+        {
+            this.animations.computeIfAbsent(type, t -> new ArrayList<>()).add(renderer -> {
+                ((LivingEntityRenderer<T, ?>) renderer).getAnimator().addAnimation(animation);
+            });
+        }
+
+        /**
+         * Gets a map of the registered animations. Used internally.
+         */
+        public Map<EntityType<?>, List<Consumer<LivingEntityRenderer<?, ?>>>> getAnimations()
+        {
+            return ImmutableMap.copyOf(animations);
```

Comment:
This would still mean the lists present in the immutable map can be modified and affect the original list, which may be undesirable. You may want to consider using a `ListMultimap` internally, and returning a (cached) view of the multimap using `Multimaps#asMap` and `Multimaps#unmodifiableListMultimap` (or `ImmutableListMultimap#copyOf` for a copy and not a view of the underlying multimap).

Reply:
Good catch. I've patched it to use `ListMultimap` and return a copy using`ImmutableListMultimap#copyOf`